### ISSUE TITLE
Haddock build failure with GHC 8

### ISSUE THREAD

**Issue by snoyberg** [aelve/microlens#72]
```
> /tmp/stackage-build9$ stack unpack microlens-0.4.4.0
Unpacked microlens-0.4.4.0 to /tmp/stackage-build9/microlens-0.4.4.0/
> /tmp/stackage-build9/microlens-0.4.4.0$ runghc -clear-package-db -global-package-db -package-db=/home/stackage/work/builds/nightly/pkgdb Setup configure --package-db=clear --package-db=global --package-db=/home/stackage/work/builds/nightly/pkgdb --libdir=/home/stackage/work/builds/nightly/lib --bindir=/home/stackage/work/builds/nightly/bin --datadir=/home/stackage/work/builds/nightly/share --libexecdir=/home/stackage/work/builds/nightly/libexec --sysconfdir=/home/stackage/work/builds/nightly/etc --docdir=/home/stackage/work/builds/nightly/doc/microlens-0.4.4.0 --htmldir=/home/stackage/work/builds/nightly/doc/microlens-0.4.4.0 --haddockdir=/home/stackage/work/builds/nightly/doc/microlens-0.4.4.0 --flags=
Configuring microlens-0.4.4.0...
> /tmp/stackage-build9/microlens-0.4.4.0$ runghc -clear-package-db -global-package-db -package-db=/home/stackage/work/builds/nightly/pkgdb Setup haddock --haddock-options=--read-interface=../base-4.9.0.0/,/home/stackage/work/builds/nightly/doc/base-4.9.0.0/base.haddock --haddock-option=--hyperlinked-source --haddock-options=--read-interface=../base-4.9.0.0/,/home/stackage/work/builds/nightly/doc/base-4.9.0.0/base.haddock --html --haddock-options=--read-interface=../base-4.9.0.0/,/home/stackage/work/builds/nightly/doc/base-4.9.0.0/base.haddock --hoogle --haddock-options=--read-interface=../base-4.9.0.0/,/home/stackage/work/builds/nightly/doc/base-4.9.0.0/base.haddock --html-location=../$pkg-$version/ --haddock-options=--read-interface=../base-4.9.0.0/,/home/stackage/work/builds/nightly/doc/base-4.9.0.0/base.haddock
Running Haddock for microlens-0.4.4.0...
Preprocessing library microlens-0.4.4.0...
Haddock coverage:
 100% ( 12 / 12) in 'Lens.Micro.Type'
  17% (  4 / 23) in 'Lens.Micro.Internal'
  Missing documentation for:
    foldring (src/Lens/Micro/Internal.hs:105)
    foldrOf (src/Lens/Micro/Internal.hs:109)
    foldMapOf (src/Lens/Micro/Internal.hs:113)
    #. (src/Lens/Micro/Internal.hs:142)
    .# (src/Lens/Micro/Internal.hs:145)
    phantom (src/Lens/Micro/Internal.hs:129)
    Each (src/Lens/Micro/Internal.hs:159)
    Index (src/Lens/Micro/Internal.hs:218)
    IxValue (src/Lens/Micro/Internal.hs:220)
    Ixed (src/Lens/Micro/Internal.hs:227)
    At (src/Lens/Micro/Internal.hs:267)
    Field1 (src/Lens/Micro/Internal.hs:320)
    Field2 (src/Lens/Micro/Internal.hs:399)
    Field3 (src/Lens/Micro/Internal.hs:438)
    Field4 (src/Lens/Micro/Internal.hs:473)
    Field5 (src/Lens/Micro/Internal.hs:504)
    Cons (src/Lens/Micro/Internal.hs:531)
    Snoc (src/Lens/Micro/Internal.hs:539)
    Strict (src/Lens/Micro/Internal.hs:547)
haddock: internal error: internal: extractDecl (ClassDecl)
CallStack (from HasCallStack):
  error, called at utils/haddock/haddock-api/src/Haddock/Interface/Create.hs:793:16 in main:Haddock.Interface.Create
```

**Comment by neongreen** [comment]
@snoyberg I've investigated this, and I believe there is no workaround other than changing the exports (basically I'd have to reexport an internal class from the main module – and even then the docs will be wonky). Could you disable haddock generation for microlens? (If I find a suitable workaround I'll open an issue on Stackage.)

**Comment by snoyberg** [comment]
I've already marked Haddocks as expected failure for microlens, so you're safe for now. When this gets resolved, please just ping this issue or the Stackage issue tracker (or send a PR even) and we'll mark the Haddocks as expected to pass so we note future regressions.

**Comment by neongreen** [comment]
Fixed in microlens-0.4.7.
